### ISSUE TITLE
can't deploy due to not being able to get available storages

### ISSUE THREAD

**Issue by morph027** [LordGaav/proxmox-deploy#9]
PVE 4.2/4.3 with different storages:

```
dir: local
    path /var/lib/vz
    content vztmpl,iso,rootdir
    maxfiles 0

dir: backup-data-iso
    path /backup-2tb/data/CD-Images
    content iso
    maxfiles 0

zfspool: vm-400g
    pool vm-400g
    sparse
    content images,rootdir

zfspool: backup-2tb-vms
    pool backup-2tb/vms
    sparse
    content rootdir,images

dir: vm-backup
    path /backup-2tb/vms/
    content backup
    maxfiles 4
```

When i run `proxmox-deploy`:

```
proxmox-deploy version 0.3 (build 7b45fa9) starting...
Asking user for configuration input
Traceback (most recent call last):
  File "/home/morph/env/bin/proxmox-deploy", line 11, in <module>
    sys.exit(main())
  File "/home/morph/env/local/lib/python2.7/site-packages/proxmoxdeploy/cli.py", line 93, in main
    (proxmox, cloudinit) = interact_with_user(args, api)
  File "/home/morph/env/local/lib/python2.7/site-packages/proxmoxdeploy/cli.py", line 77, in interact_with_user
    proxmox_answers = ask_proxmox_questions(api)
  File "/home/morph/env/local/lib/python2.7/site-packages/proxmoxdeploy/proxmox.py", line 63, in ask_proxmox_questions
    default=available_storage[0])
IndexError: list index out of range
```

**Comment by morph027** [comment]
Any help i can offer? I can provide various colorfull storage settings from my lab, if you need them to debug ;)

**Comment by Znuff** [comment]
Script seems to fail badly if your storage is ZFS.

    def get_storage(self, node):
        """
        Get available storages.

        Parameters
        ----------
        node: str
            If provided, will retrieve the storages for this specific node. If
            not, will retrieve the global storages.

        Returns
        -------
        List of storages available.
        """
        storages = []
        for storage in self.client.nodes(node).storage.get():
            if ("images" in storage['content'].split(",")
                    and storage['type'] in ("dir", "lvm", "lvmthin", "nfs")):
                storages.append(storage['storage'])
        return storages

As you can see, it checks for a storage for `images` that is dir, lvm, lvmthin or nfs, and your storage config doesn't have a matching one.

Even if you have a storage for images (`content images`) located on the ZFS pool, the script never sets the cache to Write Back, so it fails in actually starting the image.

**Comment by Znuff** [comment]
https://github.com/LordGaav/proxmox-deploy/pull/14

This adds zfspool storage support.

**Comment by morph027** [comment]
Thanks, will have a look at it and report back!

**Comment by LordGaav** [comment]
zfspool support has been merged based on the work of @Znuff , see #19
